src/pages/docs/training-certificate/[orderId]/[participantId].tsx

```diff
@@ -0,0 +1,282 @@
+import CTA from '@components/CTA';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useUser } from '@auth0/nextjs-auth0/client';
+import styled from 'styled-components';
+import { useEffect, useRef, useState } from 'react';
+import html2canvas from 'html2canvas';
+import jsPDF from 'jspdf';
+import Image from 'next/image';
+import Logo from '@assets/new/logos/brand-logo.png';
+import TopPaintIcon from '@assets/new/images/top_cert_paint.png';
+import BottomPaintIcon from '@assets/new/images/bottom_cert_paint.png';
+import { TrainingAttendanceType } from '@data/types/trainingTypes/TypeOrthoexTrainingData';
+import { formatDate } from '@utils/index';
+import { toast } from 'react-toastify';
+
+const TrainingCertificate = () => {
+	// @ts-ignore
+	const { user } = useUser();
+	const router = useRouter();
+	const pdfRef = useRef<HTMLDivElement>(null);
+
+	const [attendanceData, setAttendanceData] =
+		useState<TrainingAttendanceType | null>(null);
+	const orderId = router.query.orderId as string;
+	const participantId = router.query.participantId as string;
+
+	useEffect(() => {
```

Comment:
do this in the `getServerProps` function. That's the benefit of using next js over ordinary react

Reply:
Done ✅✅